docker/secondary-analysis-python/tools/submit/tests/test_create_envelope.py

```diff
@@ -0,0 +1,60 @@
+import unittest
+import os
+import sys
+import json
+
+#pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+#sys.path.insert(0, pkg_root)
+
+import submit.create_envelope as submit
+
+class TestCreateEnvelope(unittest.TestCase):
+
+    def test_get_entity(self):
+        with open(self.data_file('response.json')) as f:
+            js = json.load(f)
+            entity_url = submit.get_entity_url(js, 'analyses')
+            self.assertEqual(entity_url, 'http://api.ingest.dev.data.humancellatlas.org/analyses')
+
+    def test_get_input_bundle_uuid(self):
+        with open(self.data_file('analysis.json')) as f:
+            js = json.load(f)
+            self.assertEqual(submit.get_input_bundle_uuid(js), '75a7f618-9adc-48af-a249-0010305160f6')
+
+    def test_get_output_files(self):
+        with open(self.data_file('analysis.json')) as f:
+            js = json.load(f)
+            outputs = submit.get_output_files(js)
+            self.assertEqual(len(outputs), 3)
+            self.assertEqual(outputs[0]['fileName'], 'sample.bam')
+            self.assertEqual(outputs[0]['content']['name'], 'sample.bam')
+            self.assertEqual(outputs[0]['content']['format'], 'bam')
+
+    def test_check_status(self):
```

Comment:
I think this test could be split up further based on the different error codes (e.g. `test_check_status_500`). I find having multiple specific tests over one test that does multiple things helps with readability and easier debugging.

Reply:
Point taken - I split it into two, one for good and one for bad status codes.